MaxText/inference_microbenchmark.py

```diff
@@ -170,10 +170,20 @@ def collate_results(config, results, model_size, cache_size, num_model_params, i
   return results
 
 
-def write_results(results, filename):
+def write_results(results, filename, flatten_microbenchmark_results):
```

Comment:
Is the result from the original microbenchmark? It seem it's overwritten by the flatten_microbenchmark_results

Reply:
Yes, `results` is the original dictionary.

The `write_results` function will update `results` by flattening it. Currently it does not insert the flattened dict into the original `results` dict. Per your other comment, I can make the change to add it to, instead of completely replace, `results`.

Reply:
addition will be better than replace! Thanks!

Reply:
Done.